src/providers/twitch/TwitchBadges.cpp

```diff
@@ -29,25 +30,49 @@ void TwitchBadges::loadTwitchBadges()
 {
     assert(this->loaded_ == false);
 
-    QUrl url("https://badges.twitch.tv/v1/badges/global/display");
+    getHelix()->getGlobalBadges(
+        [this](auto result) {
+            auto globalBadges = HelixGlobalBadges{result};
+            auto badgeSets = this->badgeSets_.access();
 
-    QUrlQuery urlQuery;
-    urlQuery.addQueryItem("language", "en");
-    url.setQuery(urlQuery);
-
-    NetworkRequest(url)
-        .onSuccess([this](auto result) -> Outcome {
-            auto root = result.parseJson();
-
-            this->parseTwitchBadges(root);
+            for (const auto &badgeSet : globalBadges.data)
+            {
+                const auto &setID = badgeSet.setID;
+                for (const auto &version : badgeSet.versions)
+                {
+                    const auto &emote =
+                        Emote{{""},
+                              ImageSet{
+                                  Image::fromUrl(version.imageURL1x),
+                                  Image::fromUrl(version.imageURL2x, .5),
+                                  Image::fromUrl(version.imageURL4x, .25),
+                              },
+                              Tooltip{version.title},
+                              version.clickURL};
+                    (*badgeSets)[setID][version.id] =
+                        std::make_shared<Emote>(emote);
+                }
+            }
 
             this->loaded();
-            return Success;
-        })
-        .onError([this](auto res) {
-            qCWarning(chatterinoTwitch)
-                << "Error loading Twitch Badges from the badges API:"
-                << res.status() << " - falling back to backup";
+        },
+        [this](auto error, auto message) {
+            QString errorMessage("Failed to load global badges - ");
+
+            switch (error)
+            {
+                case HelixGetGlobalBadgesError::Forwarded: {
+                    errorMessage += message;
+                }
+                break;
+
+                // This would most likely happen if the service is down, or if the JSON payload returned has changed format
+                case HelixGetGlobalBadgesError::Unknown: {
+                    errorMessage += "An unknown error has occurred.";
+                }
+                break;
+            }
+            qCWarning(chatterinoTwitch) << errorMessage;
             QFile file(":/twitch-badges.json");
```

Comment:
In a follow-up PR, we can convert `twitch-badges.json` to use the new helix structure (this would allow enable a refactor of `parseTwitchBadges` that leverages the new structs from this PR).